jujugui/static/gui/src/app/components/deployment/summary/summary.js

```diff
@@ -43,6 +43,59 @@ YUI.add('deployment-summary', function() {
       validateForm: React.PropTypes.func.isRequired
     },
 
+    getInitialState: function() {
+      return {
+        regions: [],
+        activeRegion: this.props.deploymentStorage.region
+      };
+    },
+
+    componentWillMount: function() {
+      // If the model has already been deployed then we already
+      // have all the information we need.
+      var props = this.props;
+      if (props.modelCommitted) { return; }
+      // Because the user may have selected a template from the choose-cloud
+      // component we won't have any of the location data for the template
+      // to be able to properly select the correct region or cloud from the
+      // drop down list.
+      var deploymentStorage = props.deploymentStorage;
+      var templateName = deploymentStorage.templateName;
+      this._listTemplatesXHR = props.jem.listTemplates((error, credentials) => {
+        this._abortXHR('_listTemplatesXHR');
+        if (error) {
+          console.error('Unable to list templates', error);
+          return;
+        }
+        // The credentials are returned in an array so we need to loop through
+        // the list to find the one with the proper path.
+        credentials.some(credential => {
+          if (credential.path === templateName && credential.location) {
+            deploymentStorage.cloud = credential.location.cloud;
+            deploymentStorage.region = credential.location.region;
+            return true;
+          }
+        });
+        if (deploymentStorage.cloud && deploymentStorage.region) {
+          this._listRegionsXHR = props.jem.listRegions(
+            deploymentStorage.cloud, (error, regions) => {
+              this._abortXHR('_listRegionsXHR');
```

Comment:
Yep, I completely agree, for example, we currently request things like credentials/templates in several different components, when really we could just be requesting them once and passing the data down into the components.

Not a big deal, but I probably wouldn't bother with that cleanup, but if it makes you happy then leave it :).